src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoNullable.cs

```diff
@@ -79,6 +81,53 @@ protected override void OnWrite(ref WriteStackFrame current, Utf8JsonWriter writ
             }
         }
 
+        protected override void OnWriteDictionary(ref WriteStackFrame current, Utf8JsonWriter writer)
+        {
+            if (Converter == null)
+            {
+                return;
+            }
+
+            Debug.Assert(current.CollectionEnumerator != null);
+
+            string key = null;
+            TProperty? value = null;
+            if (current.CollectionEnumerator is IEnumerator<KeyValuePair<string, TProperty?>> enumerator)
+            {
+                key = enumerator.Current.Key;
+                // Avoid boxing for strongly-typed enumerators such as returned from IDictionary<string, TRuntimeProperty>
```

Comment:
I don't understand this. Where's the potential boxing that is being avoided here?